hazelcast/src/main/java/com/hazelcast/query/impl/InternalIndex.java

```diff
@@ -112,4 +115,21 @@ public interface InternalIndex extends Index {
      * @return {@code true} if the stamp is still valid, {@code false} otherwise
      */
     boolean validatePartitionStamp(long stamp);
+
+    /**
+     * @return Step-aware storages that back the Index.
+     * By default returns {@code null} that indicates the
+     * backed storage is not Step-aware.
+     */
+    default List<StepAwareStorage> getStepAwareStorages() {
+        return null;
+    }
+
+    /**
+     * @return {@code true} if the Index is backed by Tiered Store,
+     * {@code false} otherwise.
+     */
+    default boolean isBackedByTieredStore() {
```

Comment:
the same as above, my concern is that we are leaking EE features into OS

Reply:
Addressed in https://github.com/hazelcast/hazelcast-enterprise/pull/6348/commits/1ac699f3a809f14817ffa665cd426b17d5b59320